kbfsedits/tlf_history.go

```diff
@@ -313,11 +316,24 @@ func (th *TlfHistory) recomputeLocked() (
 	sort.Sort(history)
 	if len(history) > maxWritersPerHistory {
 		// Garbage-collect any writers that don't appear in the history.
+		loggedInIndex := -1
 		for i := maxWritersPerHistory; i < len(history); i++ {
+			if history[i].isLoggedIn {
+				// Don't purge the logged-in user.
+				loggedInIndex = i
+				continue
+			}
 			delete(th.byWriter, history[i].writerName)
 			delete(writersWhoNeedMore, history[i].writerName)
 		}
-		history = history[:maxWritersPerHistory]
+		if loggedInIndex > 0 {
+			// Keep the logged-in user as the last entry.
+			loggedIn := history[loggedInIndex]
+			history = history[:maxWritersPerHistory]
+			history[maxWritersPerHistory-1] = loggedIn
```

Comment:
Did you mean to do a swap here instead of duplicating `loggedIn` like what it's doing now?

Reply:
This doesn't duplicate anything, because the `for` loop above starts with `i := maxWritersPerHistory`; therefore, the loggedInIndex will always be beyond the cutoff that happens in the line above.  I'll clarify with a comment.

Reply:
I wonder if you can do:
```go
history = append(history[:maxWritersPerHistory-1], history[loggedInIndex])
```

Reply:
I probably could, though I'm not sure if that's worse performance under the hood.  But if you think it's clearer, I'm happy to change it.

Reply:
It's easier to read, and I doubt the performance will be a problem.

Reply:
I see; thanks for explaining!

`append` reuses the underlying capacity if available, I think.